### PR TITLE
Add table `azure_bastion_host`. Closes #576

### LINKED ISSUE #576: Add table azure_bastion_host
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.
Requirement - CIS v2.0.0 > 7.1 Ensure an Azure Bastion Host Exists (Automated)

[Terraform Ref](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/bastion_host)

Azure SDK-related [issue](https://github.com/Azure/azure-sdk-for-go/issues/5233), unsure if this is now supported or not, but I created this issue for rechecking purposes.

### PR DESCRIPTION
**Pull request by karanpopat**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```sql
> select
    name,
    dns_name,
    provisioning_state,
    region,
    resource_group
  from
    azure_bastion_host;
+-------------------+------------------------------------------------------------+--------------------+---------+----------------+
| name              | dns_name                                                   | provisioning_state | region  | resource_group |
+-------------------+------------------------------------------------------------+--------------------+---------+----------------+
| demo-vnet-bastion | bst-26953659-8877-453f-xxxx-b01e945537ba.bastion.azure.com | Succeeded          | westus2 | demo           |
+-------------------+------------------------------------------------------------+--------------------+---------+----------------+

> select
    h.name as bastion_host_name,
    s.id as subnet_id,
    s.name as subnet_name,
    address_prefix
  from
    azure_bastion_host h,
    jsonb_array_elements(ip_configurations) ip,
    azure_subnet s
  where
    s.id = ip -> 'properties' -> 'subnet' ->> 'id';
+-------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------+----------------+
| bastion_host_name | subnet_id                                                                                                                                                | subnet_name        | address_prefix |
+-------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------+----------------+
| demo-vnet-bastion | /subscriptions/ab6d7416-f95f-xxxx-bbb5-529d4c76659c/resourceGroups/demo/providers/Microsoft.Network/virtualNetworks/demo-vnet/subnets/AzureBastionSubnet | AzureBastionSubnet | 10.1.1.0/26    |
+-------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------+----------------+
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
Outputs:

resource_aka = "azure:///subscriptions/ab6d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest88237/providers/Microsoft.Network/bastionHosts/turbottest88237"
resource_aka_lower = "azure:///subscriptions/ab6d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest88237/providers/microsoft.network/bastionhosts/turbottest88237"
resource_dns_name = "bst-336d5248-936e-4844-ab2d-a680a2598431.bastion.azure.com"
resource_id = "/subscriptions/ab6d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest88237/providers/microsoft.network/bastionhosts/turbottest88237"
resource_name = "turbottest88237"
subscription_id = "ab6d7416-f95f-4771-bbb5-529d4c76659c"

Running SQL query: test-get-query.sql
[
  {
    "dns_name": "bst-336d5248-936e-4844-ab2d-a680a2598431.bastion.azure.com",
    "id": "/subscriptions/ab6d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest88237/providers/microsoft.network/bastionhosts/turbottest88237",
    "name": "turbottest88237",
    "resource_group": "turbottest88237",
    "tags": {
      "name": "turbottest88237"
    }
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "dns_name": "bst-336d5248-936e-4844-ab2d-a680a2598431.bastion.azure.com",
    "id": "/subscriptions/ab6d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest88237/providers/microsoft.network/bastionhosts/turbottest88237",
    "name": "turbottest88237",
    "resource_group": "turbottest88237",
    "tags": {
      "name": "turbottest88237"
    }
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "azure:///subscriptions/ab6d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest88237/providers/microsoft.network/bastionhosts/turbottest88237",
      "azure:///subscriptions/ab6d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest88237/providers/microsoft.network/bastionhosts/turbottest88237"
    ],
    "name": "turbottest88237",
    "title": "turbottest88237"
  }
]
✔ PASSED

POSTTEST: tests/azure_bastion_host

TEARDOWN: tests/azure_bastion_host

SUMMARY:

1/1 passed.
```
</details>

### CODE REVIEW

File: docs/tables/azure_bastion_host.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+```sql
+select
+	name,
+	dns_name,
+	provisioning_state,
+	region,
+	resource_group
+from
+	azure_bastion_host;
```
Comment (rajlearner17) [inline docs/tables/azure_bastion_host.md:17]: @karanpopat can you pls update the format of the query

File: azure-test/tests/azure_bastion_host/test-list-query.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+select
+  lower(id) as id,
+	name,
```
Comment (misraved) [inline azure-test/tests/azure_bastion_host/test-list-query.sql:3]: Please fix the indentation wherever needed 👍.

File: azure/table_azure_bastion_host.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+			},
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listBastionHosts,
+		},
+		Columns: azureColumns([]*plugin.Column{
+			{
+				Name:        "name",
+				Type:        proto.ColumnType_STRING,
+				Description: "The friendly name that identifies the bastion host",
```
Comment (misraved) [inline azure/table_azure_bastion_host.go:33]:
```suggestion
				Description: "The friendly name that identifies the bastion host.",
```

Comment (misraved) [reply azure/table_azure_bastion_host.go:33]: @karanpopat Please add the missing `.` at the end of column descriptions.

File: docs/tables/azure_bastion_host.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  name,
+  dns_name,
+  provisioning_state,
+  region,
+  resource_group
+from
+  azure_bastion_host;
+```
+
+### List bastion hosts that failed creation
```
Comment (misraved) [inline docs/tables/azure_bastion_host.md:20]:
```suggestion
### List bastion hosts that are in failed state
```

File: docs/tables/azure_bastion_host.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  address_prefix
+from
+  azure_bastion_host h,
+  jsonb_array_elements(ip_configurations) ip,
+  azure_subnet s
+where
+  s.id = ip -> 'properties' -> 'subnet' ->> 'id';
+```
+
+### Get ip configuration details associated with each host
```
Comment (misraved) [inline docs/tables/azure_bastion_host.md:51]:
```suggestion
### Get IP configuration details associated with each host
```

**Review by misraved — COMMENTED** [review]
@karanpopat Please take a look at the review comments. Thanks!!

**Review by misraved — APPROVED** [review]

**Pull request merged**